pkg/apis/pipeline/v1alpha1/param_types.go

```diff
@@ -30,11 +42,97 @@ type ParamSpec struct {
 	// default is set, a Task may be executed without a supplied value for the
 	// parameter.
 	// +optional
-	Default string `json:"default,omitempty"`
+	Default *ArrayOrString `json:"default,omitempty"`
 }
 
-// Param declares a value to use for the Param called Name.
-type Param struct {
+func (pp *ParamSpec) SetDefaults(ctx context.Context) {
+	if pp != nil && pp.Type == "" {
+		if pp.Default != nil {
+			// propagate the parsed ArrayOrString's type to the parent ParamSpec's type
+			pp.Type = pp.Default.Type
+		} else {
+			// ParamTypeString is the default value (when no type can be inferred from the default value)
+			pp.Type = ParamTypeString
+		}
+	}
+}
+
+// ResourceParam declares a value to use for the parameter called Name, and is used in the
+// specific context of PipelineResources.
+type ResourceParam struct {
 	Name  string `json:"name"`
 	Value string `json:"value"`
 }
+
+// Param declares an ArrayOrString to use for the parameter called name.
+type Param struct {
+	Name  string        `json:"name"`
+	Value ArrayOrString `json:"value"`
+}
+
+// ParamType indicates the type of an input parameter;
+// Used to distinguish between a single string and an array of strings.
+type ParamType string
+
+// Valid ParamTypes:
+const (
+	ParamTypeString ParamType = "string"
+	ParamTypeArray  ParamType = "array"
+)
+
+// AllParamTypes can be used for ParamType validation.
+var AllParamTypes = []ParamType{ParamTypeString, ParamTypeArray}
+
+// ArrayOrString is modeled after IntOrString in kubernetes/apimachinery:
+
+// ArrayOrString is a type that can hold a single string or string array.
+// Used in JSON unmarshalling so that a single JSON field can accept
+// either an individual string or an array of strings.
+type ArrayOrString struct {
+	Type      ParamType // Represents the stored type of ArrayOrString.
+	StringVal string
+	ArrayVal  []string
+}
+
+// String returns the single string value, or a string of concatenated
+// values from the array.
+func (arrayOrString ArrayOrString) String() string {
+	if arrayOrString.Type == ParamTypeString {
+		return arrayOrString.StringVal
+	}
+	return strings.Join(arrayOrString.ArrayVal, " ")
```

Comment:
Whereabouts is this used now?  I think you mentioned there was one spot in the code that uses this?

Reply:
Again, retyping my reply: 

You're right, that method isn't used! I removed it yesterday.